web/src/main/kotlin/com/example/ajaxproject/service/GroupChatServiceImpl.kt

```diff
@@ -30,107 +34,94 @@ class GroupChatServiceImpl (
         private val logger: Logger = LoggerFactory.getLogger(GroupChatServiceImpl::class.java)
     }
 
-    override fun createGroupRoom(createChatDto: CreateChatDto): GroupChatRoom {
-
-        val user = userService.getUserById(createChatDto.adminId)
-
-        val groupChatRoom = GroupChatRoom(
-            id = ObjectId().toHexString(),
-            chatName = createChatDto.chatName,
-            adminId = user.id,
-            chatMembers = listOf(user)
-        )
-
-        groupChatRoom.chatMembers + user
-
-        groupChatRoomRepository.save(groupChatRoom)
-
-        logger.info("Group chat room created successfully")
-
-        return groupChatRoom
-
+    override fun createGroupRoom(createChatDto: CreateChatDTO): Mono<GroupChatRoom> {
+        return userService.getById(createChatDto.adminId)
+            .flatMap { user ->
+                val groupChatRoom = GroupChatRoom(
+                    id = ObjectId().toHexString(),
+                    chatName = createChatDto.chatName,
+                    adminId = user.id,
+                    chatMembers = listOf(user)
+                )
+                groupChatRoomRepository.save(groupChatRoom)
+                    .doOnSuccess { logger.info("Group chat room created successfully") }
+            }
     }
 
-    override fun getAllChatMembers(chatId: String): List<User> {
-
-        val chatMembers = groupChatRoomRepository.findChatRoom(chatId).chatMembers
-
-        logger.info("Retrieved chat members for chat ID: {}" , chatId)
-
-        return chatMembers
+    override fun getAllChatMembers(chatId: String): Mono<List<User>> {
+        return groupChatRoomRepository.findChatRoom(chatId)
+            .flatMapMany { chatRoom ->
+                Flux.fromIterable(chatRoom.chatMembers)
+            }
+            .collectList().doOnSuccess { _ -> logger.info("Retrieved chat members for chat ID: {}", chatId) }
```

Comment:
```kotlin
            .flatMapMany { chatRoom ->
                Flux.fromIterable(chatRoom.chatMembers)
            }
```
and 
```kotlin
.collectList()
```

Doesn't make any sense!

Reply:
Will comment this conversation together with next one as it all about one method. On the start - i made it as Flux<User> and code was rather more better. But one of the recomendation from Volodymyr here https://github.com/OleksandRomaniuk/Ajax/pull/17#discussion_r1360333327 was to convert it to Mono<List> - that is why i wrote .collectList()